Kernel/Devices/HID/USBMouseDevice.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Try.h>
+#include <Kernel/Bus/USB/USBDescriptors.h>
+#include <Kernel/Bus/USB/USBRequest.h>
+#include <Kernel/Devices/DeviceManagement.h>
+#include <Kernel/Devices/HID/USBMouseDevice.h>
+#include <Kernel/KString.h>
+
+namespace Kernel {
+
+ErrorOr<NonnullRefPtr<USBMouseDevice>> USBMouseDevice::try_to_initialize(USB::Device const& usb_device)
+{
+    auto mouse_device_or_error = DeviceManagement::try_create_device<USBMouseDevice>(usb_device);
+    // FIXME: Find a way to propagate errors
+    VERIFY(!mouse_device_or_error.is_error());
+    return mouse_device_or_error.release_value();
+}
+
+USBMouseDevice::USBMouseDevice(USB::Device const& usb_device)
+    : MouseDevice()
+    , m_attached_usb_device(usb_device)
+{
+    (void)Process::create_kernel_process(m_polling_thread, MUST(KString::try_create("UHCI USB mouse")), [&] {
+        for (;;) {
+            char packet_raw[6];
+            if (!m_attached_usb_device->interrupt_pipe()) {
+                (void)Thread::current()->sleep(Time::from_milliseconds(10));
+                continue;
+            }
+
+            size_t transfer_length = MUST(m_attached_usb_device->interrupt_pipe()->interrupt_transfer(USB_REQUEST_TRANSFER_DIRECTION_DEVICE_TO_HOST | USB_REQUEST_RECIPIENT_ENDPOINT, 0, 0, 0, 6, &packet_raw));
+            if (transfer_length != 0) {
+                MousePacket packet;
+                packet.buttons = packet_raw[0] & 0x07;
+                packet.x = packet_raw[1];
+                packet.y = -packet_raw[2];
+                packet.z = packet_raw[3];
+                packet.w = 0;
+                packet.is_relative = true;
```

Comment:
Few remarks:

1. can you pass `packet_raw` to the constructor, instead of manually doing it outside? (or use a load function... of `auto packet = MousePacker::loadFromRaw(packet_raw);` ?
2. I assume you can typecase the raw packet to a `UsbMousePacket` (which is layed/packed out as needed) instead of using raw offsets. I am unsure if this is portable to ARM/32-64... You are the expert here :)